# Homework Introduction to Python

$\textbf{Ubah nama di bawah yaa!}$

- Nama: Trimitha Karina Ningsih
- IPK / Nilai Rapot: 3.57 / 9.6

In [1]:
import numpy as np

# please import as much as you need

# Normal

## 1. Operasi Variabel

> Misal kita memiliki toko sepatu. Modal membeli 1 kodi sepatu adalah Rp 1.000.000. Setiap sepatu dijual seharga Rp 80.000. Buatlah variabel modal, kuantitas, dan harga_jual sesuai informasi di atas. Hitung profit yang didapatkan jika semua sepatu berhasil terjual, dan masukkan ke variabel profit.

Saran langkah-langkah pengerjaan
1. Convert terlebih dahulu kodi dalam jumlah barang (1 kodi = xyz buah sepatu)
2. Gunakan operasi matematika sederhana untuk menghitung profit menggunakan tiga buah variabel yang sudah ditentukan

In [2]:
# Langkah 1: Konversi kodi ke jumlah sepatu
jumlah_sepatu_per_kodi = 12  # Misal 1 kodi = 12 sepatu
kodi = 1
jumlah_sepatu = kodi * jumlah_sepatu_per_kodi

# Langkah 2: Menghitung modal, kuantitas, dan harga jual
modal_per_kodi = 1000000  # Modal per kodi
modal = kodi * modal_per_kodi
harga_jual_per_sepatu = 80000  # Harga jual per sepatu
harga_jual_total = harga_jual_per_sepatu * jumlah_sepatu

# Menghitung profit
profit = harga_jual_total - modal

# Menampilkan hasil
print("Jumlah sepatu:", jumlah_sepatu)
print("Modal:", modal)
print("Harga jual total:", harga_jual_total)
print("Profit:", profit)

Jumlah sepatu: 12
Modal: 1000000
Harga jual total: 960000
Profit: -40000


Jadi, dari profit di atas, terlihat bahwa jika semua sepatu terjual dengan harga jual yang telah ditentukan, toko akan mengalami kerugian sebesar Rp 40.000. Dalam kasus ini, biaya modal dan harga jual tidak menghasilkan keuntungan yang cukup untuk menutupi modal awal yang dikeluarkan.

## 2. Package Python dan Dictionary

> Kita akan coba mengakses data scientific article dari PLOS corpus
    Menggunakan API request dengan kode berikut :
```
import requests
response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')
```
**perhatikan tanda kutipnya tidak salah ketika meng-copy ke dalam python

Pertanyaannya :
Convert response.content dari request tersebut dari JSON Format ke dalam Dictionary, dan print-kan key di dalamnya apa saja

Saran langkah-langkah pengerjaan
1. Ikuti code dari soal, kemudian ekstrak value dari variable response menggunakan dot (.) content menjadi sebuah variabel baru (dictionary)
2. Gunakan library json, convert hasil content menggunakan json.loads
3. Print keys dari dictionary tersebut


In [4]:
import requests
import json

# Langkah 1 : Ikuti code
response = requests.get(
    'http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')

# Langkah 2: Convert hasil response.content menggunakan json.loads
response_dict = json.loads(response.content)

# Langkah 3: Print keys dari dictionary tersebut
print("Keys dalam dictionary:")
for key in response_dict.keys():
    print(key)

Keys dalam dictionary:
response


Jadi, dari url link di atas, kami menyimpulkan bahwa memiliki keys "response"

# Intermediate

## 3. Operasi dalam List

> Buatlah list dengan 10 nama kota di indonesia, kemudian:
1. Urutkan nama kota di dalam list berdasarkan abjad (a)
2. Ambil 4 kota pertama dari list tersebut (yang telah diurutkan) (b)
3. Ambil 3 kota terakhir dari list tersebut (yang telah diurutkan) (c)

Saran langkah-langkah pengerjaan
1. Tidak ada ketentuan nama kota (misal berdasarkan provinsi X atau pulau Y), dibebaskan untuk memilih. Buat nama-nama kota tersebut dalam sebuah list.
2. Gunakan fungsi sorted(X) untuk mengurutkan secara kecil-ke-besar (sesuai abjad), dimana X adalah contoh variabel list pada langkah (a)
3. Gunakan hasil poin (a), urutkan nama kota di dalam list berdasarkan abjad, untuk mengerjakan poin (b) dan (c).

In [6]:
# Langkah 1: Membuat list nama kota
nama_kota = ["Jakarta", "Surabaya", "Bandung", "Medan", "Semarang",
             "Makassar", "Yogyakarta", "Palembang", "Balikpapan", "Denpasar"]

# Langkah 2: Mengurutkan nama kota berdasarkan abjad
nama_kota_urut = sorted(nama_kota)

# Langkah 3: Mengambil 4 kota pertama dan 3 kota terakhir
kota_pertama = nama_kota_urut[:4]
kota_terakhir = nama_kota_urut[-3:]

# Menampilkan hasil
print("1. Nama kota diurutkan berdasarkan abjad:", nama_kota_urut)
print("2. 4 kota pertama:", kota_pertama)
print("3. 3 kota terakhir:", kota_terakhir)

1. Nama kota diurutkan berdasarkan abjad: ['Balikpapan', 'Bandung', 'Denpasar', 'Jakarta', 'Makassar', 'Medan', 'Palembang', 'Semarang', 'Surabaya', 'Yogyakarta']
2. 4 kota pertama: ['Balikpapan', 'Bandung', 'Denpasar', 'Jakarta']
3. 3 kota terakhir: ['Semarang', 'Surabaya', 'Yogyakarta']


## 4. Function dalam List

> Team sales memiliki list total transaksi dari 10 merchant pada bulan Juni 2020, yaitu:
```
total_transaksi = [ 19, 35, 23, 25, 10, 15, 33, 45, 21, 30 ]
```
1. Trx minimum
2. Rata-rata
3. Trx maksimal

Saran langkah-langkah pengerjaan
1. Copy variabel total_transaksi yang ada di soal ke notebook kita (atau ketik ulang boleh)
2. Kemudian, hitung menggunakan function operasi dalam list yang telah dipelajari di kelas pada topik list. Print hasilnya setiap poin (a), (b), dan (c)

In [5]:
# Langkah 1 : Copy variabel
total_transaksi = [19, 35, 23, 25, 10, 15, 33, 45, 21, 30]

# Langkah 2: Menghitung nilai minimum, rata-rata, dan maksimal
trx_minimum = min(total_transaksi)
trx_rata_rata = sum(total_transaksi) / len(total_transaksi)
trx_maksimal = max(total_transaksi)

# Langkah 3: Menampilkan hasil
print("1. Transaksi minimum:", trx_minimum)
print("2. Rata-rata transaksi:", trx_rata_rata)
print("3. Transaksi maksimal:", trx_maksimal)


1. Transaksi minimum: 10
2. Rata-rata transaksi: 25.6
3. Transaksi maksimal: 45


Jadi, dari hasil operasi list di atas, dapat disimpulkan bahwa nilai minimal yaitu sebesar 10, max sebesar 45 dan rata-rata 25.6

# Advanced

## 5. NumPy Array

> Buatlah array dari 5 bilangan ganjil pertama (dimulai dari 1), Kemudian lakukan perintah berikut :
1. Kali-kan array tersebut dengan 10
2. Bagi-kan array (hasil dari bagian a) dengan 3
3. Kali-kan array (hasil dari bagian b) dengan array baru :
`[-45, 43, 3, 56, 5]`
4. Carilah pada index ke-berapa nilai tertinggi dan terendah dari array (hasil dari bagian c)

Saran langkah-langkah pengerjaan
1. HATI-HATI dalam pengerjaan, karena apabila poin sebelumnya salah, maka selanjutnya akan kebawa salah (misal (b) salah, maka poin (c) sampai (d) akan salah)
2. Untuk array 5 bilangan ganjil pertama boleh manual boleh menggunakan fungsi, dimulai dari 1
3. Ikuti perintah soal, gunakan numpy array, jangan list. Sesuaikan dengan permintaan soal pada setiap butir.



In [ ]:
# your code goes here!


Jadi, dari hasil operasi numpy di atas, dapat disimpulkan bahwa...

# Terimakasih, salam dari ... (asal daerah Anda)